filebeat/tests/system/test_autodiscover.py

```diff
@@ -64,41 +44,47 @@ def test_default_settings(self):
         """
         Test docker autodiscover default config settings
         """
-        import docker
-        docker_client = docker.from_env()
-
-        self.render_config_template(
-            inputs=False,
-            autodiscover={
-                'docker': {
-                    'cleanup_timeout': '0s',
-                    'hints.enabled': 'true',
-                    'hints.default_config': '''
-                      type: log
-                      paths:
-                        - %s/${data.container.image}.log
-                    ''' % self.working_dir,
+        with self.container_running() as container:
+            self.render_config_template(
+                inputs=False,
+                autodiscover={
+                    'docker': {
+                        'cleanup_timeout': '0s',
+                        'hints.enabled': 'true',
+                        'hints.default_config': '''
+                          type: log
+                          paths:
+                            - %s/${data.container.name}.log
+                        ''' % self.working_dir,
+                    },
                 },
-            },
-        )
+            )
+
+            self._test(container)
 
-        with open(os.path.join(self.working_dir, 'busybox.log'), 'wb') as f:
+    def _test(self, container):
+        with open(os.path.join(self.working_dir, f'{container.name}.log'), 'wb') as f:
             f.write(b'Busybox output 1\n')
 
         proc = self.start_beat()
-        docker_client.images.pull('busybox')
-        docker_client.containers.run('busybox', 'sleep 1')
-
         self.wait_until(lambda: self.log_contains('Starting runner: input'))
-        self.wait_until(lambda: self.log_contains('Stopping runner: input'))
+        self.wait_until(lambda: self.output_has(lines=1))
```

Comment:
Just out of curiosity :) why we remove the `log_contains('Stopping runner: input')` check?

Reply:
I thought it was not relevant for the test. But I can add it again after the container is stopped. I will give it a try and to importing docker from the head of the file.